source/postcard-c/postcard.h

```diff
@@ -0,0 +1,1184 @@
+postcard_error_t postcard_encode_i32(postcard_slice_t* slice, int32_t value);
+/// Encodes a u64 to the passed in slice.
+///
+/// *Arguments*
+/// - slice - a pointer to an initialized `postcard_slice_t`
+/// - value - the value to encode
+///
+/// *Side Effects / Return*
+/// If there is not enough room in the buffer `postcard_error_t` will return a
+/// non-zero value If encoding was successful, slice.len will be incremented by
+/// the number of encoded bytes
+postcard_error_t postcard_encode_u64(postcard_slice_t* slice, uint64_t value);
+/// Encodes a i64 to the passed in slice.
+///
+/// *Arguments*
+/// - slice - a pointer to an initialized `postcard_slice_t`
+/// - value - the value to encode
+///
+/// *Side Effects / Return*
+/// If there is not enough room in the buffer `postcard_error_t` will return a
+/// non-zero value If encoding was successful, slice.len will be incremented by
+/// the number of encoded bytes
+postcard_error_t postcard_encode_i64(postcard_slice_t* slice, int64_t value);
+/// Encodes a f32 to the passed in slice.
+///
+/// *Arguments*
+/// - slice - a pointer to an initialized `postcard_slice_t`
+/// - value - the value to encode
+///
+/// *Side Effects / Return*
+/// If there is not enough room in the buffer `postcard_error_t` will return a
+/// non-zero value If encoding was successful, slice.len will be incremented by
+/// the number of encoded bytes
+postcard_error_t postcard_encode_f32(postcard_slice_t* slice, float value);
+
+/// Encodes a f64 to the passed in slice.
+///
+/// *Arguments*
+/// - slice - a pointer to an initialized `postcard_slice_t`
+/// - value - the value to encode
+///
+/// *Side Effects / Return*
+/// If there is not enough room in the buffer `postcard_error_t` will return a
+/// non-zero value If encoding was successful, slice.len will be incremented by
+/// the number of encoded bytes
+postcard_error_t postcard_encode_f64(postcard_slice_t* slice, double value);
+
+/// Encodes a byte array to the passed in slice.
+///
+/// *Arguments*
+/// - slice - a pointer to an initialized `postcard_slice_t`
+/// - bytes - a pointer to the byte array to encode
+/// - length - the length of the byte array
+/// *Safety*
+/// The user must ensure that bytes points to a valid initialized memory, and
+/// that length is not greater than the size of the array.
+///
+/// *Side Effects /
+/// Returns* If there is not enough room in the buffer `postcard_error_t` will
+/// return a non-zero value If encoding was successful, slice.len will be
+/// incremented by the number of encoded bytes
+postcard_error_t postcard_encode_byte_array(postcard_slice_t* slice,
+    const uint8_t* bytes,
+    size_t length);
+
+/// Encodes a string to the passed in slice.
+///
+/// *Arguments*
+/// - slice - a pointer to an initialized `postcard_slice_t`
+/// - string - a pointer to the string to encode. This string should be utf8
+/// encoded string. It shouldn't be null-terminated.
+/// - length - the length of the string
```

Comment:
To make it more clear that this is talking about each byte to be written and not the number of codepoints.

```suggestion
/// - length - the length of the string in bytes
```